utils/include/CJSONFormatSaver.cpp

```diff
@@ -4,31 +4,67 @@
 #include "CJSONFormatter.h"
 #include "CJSONFormatSaver.h"
 
+CJSONFormatSaver::CJSONFormatSaver(const std::string& path_to_file) :
+	m_path_to_file(path_to_file), m_number_of_spaces(3)
+{ };
+
 bool CJSONFormatSaver::TrySaveToFile(CJSONFormatter& formatted_data)
 {
-	std::ifstream json_file_to_read;
-	json_file_to_read.open(m_path_to_file);
-	nlohmann::json buff;
-	if (Utils::IsFileEmpty(json_file_to_read))
+	if (Utils::TryCreateFileIfNotExist(m_path_to_file))
+	{
+		//write to log??
+	}
+
+	std::ofstream JSON_file_to_save(m_path_to_file, std::ios::binary || 
+									std::ios::app);
+
+	if (!JSON_file_to_save.is_open())
 	{
-		buff = *formatted_data.GetJSONFormattedData();
+		JSON_file_to_save.close();
+		return false;
 	}
-	else
+	if (Utils::IsFileEmpty(JSON_file_to_save))
 	{
-		json_file_to_read >> buff;
+		if (TryWriteToFile(JSON_file_to_save, formatted_data))
+		{
+			JSON_file_to_save.close();
+			return true;
+		}
+		JSON_file_to_save.close();
+		return false;
 	}
-	json_file_to_read.close();
-	if (!buff.is_null())
+	JSON_file_to_save.seekp(0, JSON_file_to_save.end);
+
+	std::streampos position = JSON_file_to_save.tellp();
+	constexpr unsigned short num_of_bities_to_last_data = 5;
+
+	JSON_file_to_save.seekp(position - 
+		static_cast<std::streampos>(num_of_bities_to_last_data));
+
+	position = JSON_file_to_save.tellp();
+	if (!TryWriteToFile(JSON_file_to_save, formatted_data))
 	{
-		buff.merge_patch(*formatted_data.GetJSONFormattedData());
+		JSON_file_to_save.close();
+		// exception
+		return false;
 	}
-	std::ofstream JSON_file_to_save(m_path_to_file);
+	constexpr char symbol_instead_brace[] = { ',' };
+	
+	JSON_file_to_save.seekp(position);
+	JSON_file_to_save.write(symbol_instead_brace, 1);
```

Comment:
Can't you use `const char symbol_instead_brace = ',';` and `JSON_file_to_save.write(&symbol_instead_brace, 1);` instead?